CMakeLists.txt

```diff
@@ -610,6 +641,26 @@ if(UA_ENABLE_DISCOVERY)
     list(APPEND lib_sources ${PROJECT_SOURCE_DIR}/src/server/ua_discovery_manager.c)
 endif()
 
+if(UA_ENABLE_GDS)
+    list(APPEND exported_headers ${PROJECT_SOURCE_DIR}/src/gds/ua_record_datatype.h)
+    list(APPEND internal_headers ${PROJECT_SOURCE_DIR}/src/gds/ua_registration_manager.h)
+    list(APPEND lib_sources ${PROJECT_SOURCE_DIR}/src/gds/ua_gds_ns.c
+                            ${PROJECT_SOURCE_DIR}/src/gds/ua_registration_manager.c)
+    if(UA_ENABLE_GDS_CM)
+        #has to be inserted before ua_server_config.h
+        list(INSERT exported_headers 18 ${PROJECT_SOURCE_DIR}/include/ua_plugin_ca.h)
+        list(APPEND internal_headers ${PROJECT_SOURCE_DIR}/src/gds/ua_certificate_manager.h)
```

Comment:
While working on this branch we have found out issues with respect to build. The below change will fix it.

list(INSERT internal_headers 22 ${PROJECT_SOURCE_DIR}/src/gds/ua_certificate_manager.h)